piet-wgsl/shader/shared/blend.wgsl

```diff
@@ -45,23 +45,23 @@ fn color_burn(cb: f32, cs: f32) -> f32 {
 }
 
 fn hard_light(cb: vec3<f32>, cs: vec3<f32>) -> vec3<f32> {
-    return mix(
+    return select(
         screen(cb, 2.0 * cs - 1.0),
         cb * 2.0 * cs,
-        vec3<f32>(cs <= vec3<f32>(0.5))
+        cs <= vec3(0.5)
     );
 }
 
 fn soft_light(cb: vec3<f32>, cs: vec3<f32>) -> vec3<f32> {
-    let d = mix(
+    let d = select(
         sqrt(cb),
-        ((16.0 * cb - vec3(12.0)) * cb + vec3(4.0)) * cb,
-        vec3<f32>(cb <= vec3<f32>(0.25))
+        ((16.0 * cb - 12.0) * cb + 4.0) * cb,
+        cb <= vec3(0.25)
     );
-    return mix(
-        cb + (2.0 * cs - vec3(1.0)) * (d - cb),
-        cb - (vec3(1.0) - 2.0 * cs) * cb * (vec3(1.0) - cb),
-        vec3<f32>(cs <= vec3<f32>(0.5))
+    return select(
+        cb + (2.0 * cs - 1.0) * (d - cb),
+        cb - (1.0 - 2.0 * cs) * cb * (1.0 - cb),
+        (cs <= vec3(0.5))
```

Comment:
If this doesn't already have inlay hints, it probably needs them

It's not immediately clear if this is choosing the first argument or the second from (mix has a similar issue, fwiw)